src/main/java/io/cdap/plugin/successfactors/common/util/ExceptionParser.java

```diff
@@ -0,0 +1,173 @@
+    throws SuccessFactorsServiceException {
+
+    String failureMessage = msg;
+    SuccessFactorsError error = null;
+
+    if (responseContainer.getHttpStatusCode() == HttpURLConnection.HTTP_UNAUTHORIZED) {
+      failureMessage += ResourceConstants.ROOT_CAUSE_LOG.getMsgForKey()
+        .concat(ResourceConstants.ERR_INVALID_CREDENTIAL.getMsgForKey());
+
+      throw new SuccessFactorsServiceException(failureMessage, responseContainer.getHttpStatusCode());
+    } else if (responseContainer.getHttpStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+      failureMessage += ResourceConstants.ROOT_CAUSE_LOG.getMsgForKey()
+        .concat(ResourceConstants.ERR_INVALID_ENTITY_NAME.getMsgForKey());
+
+      throw new SuccessFactorsServiceException(failureMessage, responseContainer.getHttpStatusCode());
+    } else if (responseContainer.getHttpStatusCode() != HttpURLConnection.HTTP_OK) {
+      String rawResponseString = "";
+      InputStream rawStream = responseContainer.getResponseStream();
+      if (rawStream != null) {
+        try (BufferedReader bufferedReader = new BufferedReader(
+          new InputStreamReader(rawStream, StandardCharsets.UTF_8))) {
+
+          rawResponseString = bufferedReader.lines().collect(Collectors.joining(" "));
+          if (rawResponseString.contains("Refer to https")) {
+            rawResponseString = rawResponseString.substring(0, rawResponseString.indexOf("Refer to https"));
+          }
+        } catch (IOException ioe) {
+          // no-ops
+        }
+      }
+
+      LOG.error("HTTP Code: {}", responseContainer.getHttpStatusCode());
+      LOG.error("Detailed error message: {} {}", msg, rawResponseString);
+
+      try {
+        error = GSON.fromJson(rawResponseString, SuccessFactorsError.class);
+      } catch (JsonSyntaxException | JsonIOException je) {
+        // html errors are only found in case of invalid SuccessFactors service namespace
+        if (rawResponseString.startsWith("<html>")) {
+          failureMessage += ResourceConstants.ERR_INVALID_ENTITY_NAME.getMsgForKey();
+          throw new SuccessFactorsServiceException(failureMessage, SuccessFactorsServiceException.INVALID_SERVICE);
+        }
+
+        // this exception may occur when the rawResponseString contains non-json format such as text | html and in
+        // OData it is possible to receive text | html format response based on the type of errors.
+        failureMessage += ResourceConstants.ROOT_CAUSE_LOG.getMsgForKey().concat(rawResponseString);
+      }
+      throw new SuccessFactorsServiceException(failureMessage, responseContainer.getHttpStatusCode(), error);
+    }
+
+    if (SuccessFactorsUtil.isNullOrEmpty(responseContainer.getDataServiceVersion())) {
+      failureMessage += ResourceConstants.ERR_MISSING_DATASERVICE_VERSION.getMsgForKey();
+      throw new SuccessFactorsServiceException(failureMessage, NO_VERSION_FOUND);
+    }
+
+    if (!responseContainer.getDataServiceVersion().equals("2.0")) {
```

Comment:
use constant for version 2.0

Reply:
Added constant `SUPPORTED_DATASERVICE_VERSION` 444c10c0add0efdb6799cd258c03b10d1f479f5e